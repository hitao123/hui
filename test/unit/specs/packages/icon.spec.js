import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import Icon from '../../../../packages/icon';
import IconDemo from '../../../../docs/src/views/icon';

describe('Icon component', () => {
  it('should show Icon', () => {
    const warpper = shallowMount(Icon, {
      propsData: {
        name: 'add'
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should show IconDemo', () => {
    expect(renderToString(IconDemo)).toMatchSnapshot();
  });
});
