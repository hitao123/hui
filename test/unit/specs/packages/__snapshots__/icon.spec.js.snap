// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should show Icon 1`] = `<i class="h-icon h-icon-add"></i>`;

exports[`Icon component should show IconDemo 1`] = `
<div class="demo-icon"><i class="h-icon h-icon-add-o" style="color:undefined;font-size:32px;"></i> <i class="h-icon h-icon-alipay" style="color:undefined;font-size:32px;"></i> <i class="h-icon h-icon-wechat" style="color:undefined;font-size:32px;"></i>
  <i class="h-icon h-icon-add" style="color:undefined;font-size:32px;"></i> <i class="h-icon h-icon-add2" style="color:undefined;font-size:32px;"></i> <i class="h-icon h-icon-arrow-left" style="color:undefined;font-size:32px;"></i> <i class="h-icon h-icon-close"
    style="color:undefined;font-size:32px;"></i> <i class="h-icon h-icon-passed" style="color:undefined;font-size:32px;"></i> <i class="h-icon h-icon-info-o" style="color:#f00;font-size:32px;"></i></div>
`;
